File: programmers/lv2/연속_부분_수열_합의_개수.py
def solution(elements):
    answer = set()

    for i in range(len(elements)):
        t = [elements[i]]
        cnt = 0
        while cnt != len(elements)-1:
            i += 1
            t.append(t[-1]+elements[i%len(elements)])
            cnt += 1
        answer.update(t)
    
    return len(answer)

File: programmers/lv2/연속_부분_수열_합의_개수.js
function solution(elements) {
  const sumSet = new Set();
  for (let i = 1; i <= elements.length; i++) {
    for (let j = 0; j < elements.length; j++) {
      let sum = 0;
      for (let k = 0; k < i; k++) {
        sum += elements[(j + k) % elements.length];
      }
      sumSet.add(sum);
    }
  }
  return sumSet.size;
}
